### PR TITLE
Cleanups and enable MMU test

### PR DESCRIPTION
**Pull request by stffrdhrn**
Various whitespace and duplicate code cleanups.
Fix the Makefile so it actually tracks dependencies.

The MMU test seems completely broken
  - Make it check tlb sets and ways from the CPU not hard coded
  - Fix compiler warnings
  - Fix logic for bss/stack being in different places it assumed everything was in the start of memory

### CODE REVIEW

File: native/or1200/or1200-cbasic.c [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /* Test basic c functionality, and BSS section clearing.  */
-#include "cpu-utils.h"
-#include "lib-utils.h"
+//#include "cpu-utils.h"
```
Comment (olofk) [inline native/or1200/or1200-cbasic.c:2]: Remove the commented lines completely perhaps?

**Comment by olofk** [comment]
I took a quick look, and other than a couple of commented lines I couldn't find anything controversial so I trust you got this right

File: native/or1200/or1200-cbasic.c (hunk 1 above)
Comment (stffrdhrn) [reply native/or1200/or1200-cbasic.c:2]: Yes, I will. This old code is pretty messy.  Once it's all working I'll do a once over cleanup.

Thanks for having a look. There is more to come for the mmu test which is still broken.

**Comment by stffrdhrn** [comment]
Note, I started working on this cleanup a while back and stopped.  Now I need to test traps on mor1kx as I suspect a bug is causing the debugger to fall. It may be fixed by this pr but I need a test bench to be sure. 

https://github.com/openrisc/mor1kx/pull/108

**Comment by stffrdhrn** [comment]
ok I messed up merging this. Fixed it manually

**Pull request merged**